<!DOCTYPE html>
<!-- DO NOT EDIT! This test has been generated by /html/canvas/tools/gentest.py. -->
<meta charset="UTF-8">
<title>Canvas test: 2d.imageData.create2.this</title>
<script src="/resources/testharness.js"></script>
<script src="/resources/testharnessreport.js"></script>
<script src="/html/canvas/resources/canvas-tests.js"></script>
<link rel="stylesheet" href="/html/canvas/resources/canvas-tests.css">
<body class="show_output">

<h1>2d.imageData.create2.this</title>
<script src="/resources/testharness.js"></script>
<script src="/resources/testharnessreport.js"></script>
<script src="/html/canvas/resources/canvas-tests.js"></script>
<link rel="stylesheet" href="/html/canvas/resources/canvas-tests.css">
<body class="show_output">

<h1>2d.imageData.create2.this</h1>
<p class="desc">createImageData(sw, sh) should throw when called with the wrong |this|</p>

<p class="notes">Defined in "Web IDL" (draft)
<script>
var t = async_test("createImageData(sw, sh) should throw when called with the wrong |this|</p>

<p class="notes">Defined in "Web IDL" (draft)
<script>
var t = async_test("createImageData(sw, sh) should throw when called with the wrong |this|");
_addTest(function(canvas, ctx) {

  assert_throws_js(TypeError, function() { CanvasRenderingContext2D.prototype.createImageData.call(null, 1, 1); });
  assert_throws_js(TypeError, function() { CanvasRenderingContet2D.prototype.createImageData.call(undefined, 1, 1); });
  assert_throws_js(TypeError, function() { CanvasRenderingContext2D.prototype.createImageData.createImageData.call(undefined, 1, 1); });
  assert_throws_js(TypeError, function() { CanvasRenderingContext2D.prototype.createImageData.call({}, 1, 1); });

});
</script>
<canvas id="c" class="output" width="100" height="50"><p class="fallback">FAIL (fallback content)</p></canvas>

<ul id="d"></ul>
<p class="output"><p class="output"><p class="output"><p class="output"><p class="output"><p class="output"><p class="output"><p class="output"><p class="output"><p class="output"><p class="output"><p class="output"><p class="output"><p class="output"><p class="output"><p class="output"><p class="output"><p class="output"><p class="output"><p class="output"><p class="output"><p class="output"><p class="output"><p class="output"><p class="output"><p class="output"><p class="output"><p class="output"><p class="output"><p class="output"><p class="output"><p class="output"><p class="output"><p class="output"><p class="output"><p class="output"><p class="output"><p class="output"><p class="output"><p class="output"><p class="output"><p class="output"><p class="output"><p class="output"><p class="output"><p class="output"><p class="output"><p class="output"><p class="output"><p class="output"><p class="output"><p class="output"><p class="output"><p class="output"><p class="output"><p class="output"><p class="output"><p class="output"><p class="output"><p class="output"><p class="output"><p class="output"><p class="output"><p class="output"><p class="output">Actual output:</p></p></p></p></p></p></p></p></p></p></p></p></p></p></p></p></p></p></p></p></p></p></p></p></p></p></p></p></p></p><<link href="/html/canvas/resources/canvas-tests.css">
<body class="show_output">

<h1>2d.imageData.create2.this</h1>
<p class="desc">createImageData(sw, sh) should throw when called with the wrong |this|</p>

<p class="notes">Defined in "Web IDL" (draft)
<script>
var t = async_test("createImageData(sw, sh) should throw when called with the wrong |this|");
_addTest(function(canvas, ctx) {

  assert_throws_js(TypeError, function() { CanvasRenderingContext2D.prototype.createImageData.call(null, 1, 1); });
  assert_throws_js(TypeError, function() { CanvasRenderingContet2D.meta charset="UTF-8"><body><body>

